File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int bitMask = 0;
        for(var x: allowed.toCharArray()) {
                int temp = 1;
                int shift = x - 'a';
                temp = temp<<shift;
                bitMask = bitMask|temp;
        }
        int count = 0;
        for(var x: words) {
            boolean isConsistent = true;
            for(var ch: x.toCharArray()) {
                    int temp = 1;
                    int shift = ch - 'a';
                    temp = temp<<shift;
                    if((temp&bitMask)==0) {
                        isConsistent = false;
                        break;
                    }  
            }
            if(isConsistent) count++;
        }
        return count;
    }
}

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        bool allowedChars[26] = {false};
        for (char c : allowed) {
            allowedChars[c - 'a'] = true;
        }

        int count = 0;
        for (const string& word : words) {
            bool isConsistent = true;
            for (char c : word) {
                if (!allowedChars[c - 'a']) {
                    isConsistent = false;
                    break;
                }
            }
            if (isConsistent) count++;
        }

        return count;
    }
};
